### PR TITLE
[WIP] Feature/contao4

### PR DESCRIPTION
**Pull request by baumannsven**
This is for the new release and works with Contao 4.4

### CODE REVIEW

File: src/Resources/config/services.yml [hunk 1]
```diff
@@ -1 +1,10 @@
 services:
+    ## The services block
```
Comment (richardhj) [inline src/Resources/config/services.yml]: Should the services here get a `cca` prefix too? 🤔

File: src/DC/General.php [hunk 2]
```diff
@@ -205,9 +206,42 @@ protected function getTablenameCallback($strTable)
...
     public function __get($name)
     {
+        $environment          = $this->getEnvironment();
+        $inputProvider        = $environment->getInputProvider();
+        $dataDefinition       = $environment->getDataDefinition();
+        $modelRelation        = $dataDefinition->getModelRelationshipDefinition();
+        $parentDataDefinition = $environment->getParentDataDefinition();
+
         switch ($name) {
+            case 'id':
```
Comment (baumannsven) [inline src/DC/General.php]: Can see here:
https://github.com/contao-community-alliance/dc-general/pull/352

File: src/Resources/config/services.yml (hunk 1 above)
Comment (baumannsven) [inline src/Resources/config/services.yml]: Yes it´s right. I have do this

**Comment by discordier** [comment]
@baumannsven Somehow this PR is now severely messed up, could you merge master and resolve the conflicts?

**Comment by baumannsven** [comment]
@discordier I have rebase this branch to the master branch

**Comment by richardhj** [comment]
@baumannsven Can you please allow `^4.0` for the symfony packages in the `composer.json`?
```json
        "symfony/event-dispatcher": "^3.0",
        "symfony/expression-language": "^3.0"
```

It is unable to install with Contao 4.6-stable otherwise.

**Comment by baumannsven** [comment]
@discordier Can i implement in the RequestScopeDeterminator a function for detecting symfony is in develop mode? Or you have a better idea?

**Comment by baumannsven** [comment]
@richardhj  I have bump the requirements for symfony 4, so you can use DCG in CTO 4.6

```
contao-community-alliance/dependency-container
contao-community-alliance/event-dispatcher
contao-community-alliance/events-contao-bindings
contao-community-alliance/translator
```

**Comment by baumannsven** [comment]
I close this pull request. This pull request becomes newer no commits and is out of date. The actual pull request is #456

**Pull request closed without merging**